app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -23,6 +23,43 @@ def send_code
 
     private
 
+    def aal3_requirement_redirect
+      aal3_url = aal3_redirect_url
+      if aal3_url
+        redirect_to aal3_url
+      elsif AAL3Policy.new(session).aal3_required? && user_fully_authenticated?
```

Comment:
Since ApplicationController now has a memoized `aal3_policy` we can use that here instead of a new object right?